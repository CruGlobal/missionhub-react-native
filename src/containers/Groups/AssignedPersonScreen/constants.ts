export const CONTACT_PERSON_SCREEN = 'nav/CONTACT_PERSON';
export const IS_USER_CREATED_MEMBER_PERSON_SCREEN =
  'nav/IS_USER_CREATED_MEMBER_PERSON';
export const IS_GROUPS_MEMBER_PERSON_SCREEN = 'nav/IS_GROUPS_MEMBER_PERSON';
export const MEMBER_PERSON_SCREEN = 'nav/MEMBER_PERSON';
export const ME_PERSONAL_PERSON_SCREEN = 'nav/ME_PERSONAL_PERSON';
export const IS_GROUPS_ME_COMMUNITY_PERSON_SCREEN =
  'nav/IS_GROUPS_ME_COMMUNITY_PERSON';
export const ME_COMMUNITY_PERSON_SCREEN = 'nav/ME_COMMUNITY_PERSON';
