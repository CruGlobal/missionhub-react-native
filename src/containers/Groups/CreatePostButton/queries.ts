import gql from 'graphql-tag';

import { AVATAR_FRAGMENT } from '../../../components/Avatar/queries';

export const GET_MY_COMMUNITY_PERMISSION_QUERY = gql`
  query getMyCommunityPermission($id: ID!, $myId: ID!) {
    community(id: $id) {
      id
      people(personIds: [$myId]) {
        edges {
          communityPermission {
            permission
          }
        }
        nodes {
          ...Avatar
        }
      }
    }
  }
  ${AVATAR_FRAGMENT}
`;
