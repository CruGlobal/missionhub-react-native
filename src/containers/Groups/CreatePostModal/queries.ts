import gql from 'graphql-tag';

export const GET_MY_COMMUNITY_PERMISSION_QUERY = gql`
  query getMyCommunityPermission($id: ID!, $myId: ID!) {
    community(id: $id) {
      id
      people(personIds: [$myId]) {
        edges {
          communityPermission {
            permission
          }
        }
      }
    }
  }
`;
