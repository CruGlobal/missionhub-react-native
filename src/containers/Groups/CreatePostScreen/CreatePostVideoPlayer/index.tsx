import React, { useState } from 'react';
import { View } from 'react-native';

import TrashIcon from '../../../../../assets/images/trashIcon.svg';
import PlayButton from '../../../../../assets/images/playIcon.svg';
import { Touchable } from '../../../../components/common';
import VideoPlayer from '../../../../components/VideoPlayer';
import theme from '../../../../theme';

import styles from './styles';

interface CreatePostVideoPlayerProps {
  uri: string;
  onDelete: () => void;
}

export const CreatePostVideoPlayer = ({
  uri,
  onDelete,
}: CreatePostVideoPlayerProps) => {
  const [paused, setPaused] = useState(true);

  const togglePaused = () => setPaused(!paused);

  const renderVideoPlayerControls = () => (
    <View
      style={{
        flex: 1,
        flexDirection: 'column',
        justifyContent: 'flex-start',
        alignItems: 'center',
      }}
    >
      <View
        style={{
          width: theme.fullWidth,
          flexDirection: 'row',
          justifyContent: 'flex-end',
          alignItems: 'center',
        }}
      >
        <Touchable
          testID="DeleteButton"
          onPress={onDelete}
          // @ts-ignore
          type="transparent"
          style={styles.deleteButton}
        >
          <TrashIcon color={theme.white} height={24} width={24} />
        </Touchable>
      </View>
      <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}>
        <Touchable style={styles.playButton} onPress={togglePaused}>
          <PlayButton />
        </Touchable>
      </View>
    </View>
  );

  return (
    <VideoPlayer
      uri={uri}
      customControls={renderVideoPlayerControls()}
      paused={paused}
      width={theme.fullWidth}
    />
  );
};
