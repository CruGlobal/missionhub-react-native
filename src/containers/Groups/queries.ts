import gql from 'graphql-tag';

const COMMUNITY_FRAGMENT = gql`
  fragment CommunityFragment on Community {
    id
    name
    unreadCommentsCount
    communityPhotoUrl
    owner: people(permissions: [owner]) {
      nodes {
        firstName
        lastName
      }
    }
    report(period: "P1W") {
      contactCount
      memberCount
      unassignedCount
    }
  }
`;

export const GET_COMMUNITIES_QUERY = gql`
  query GetCommunities($communityCursor: String) {
    globalCommunity {
      usersReport {
        usersCount
      }
    }
    communities(
      ministryActivitiesOnly: true
      sortBy: name_ASC
      first: 10
      after: $communityCursor
    ) {
      nodes {
        ...CommunityFragment
      }
      pageInfo {
        endCursor
        hasNextPage
      }
    }
  }
  ${COMMUNITY_FRAGMENT}
`;
