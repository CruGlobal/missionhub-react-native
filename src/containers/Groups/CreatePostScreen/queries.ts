import gql from 'graphql-tag';

import { COMMUNITY_FEED_ITEM_FRAGMENT } from '../../../components/CommunityFeedItem/queries';

export const CREATE_POST = gql`
  mutation CreatePost(
    $input: CreatePostInput!
    $commentsCursor: String # not used by this query but needed to make CommunityFeedItemCommentLike.comments fragment happy
  ) {
    createPost(input: $input) {
      post {
        id
        feedItem {
          read
          ...CommunityFeedItem
        }
      }
    }
  }
  ${COMMUNITY_FEED_ITEM_FRAGMENT}
`;

export const UPDATE_POST = gql`
  mutation UpdatePost($input: UpdatePostInput!) {
    updatePost(input: $input) {
      post {
        id
        content
        mediaExpiringUrl
      }
    }
  }
`;
