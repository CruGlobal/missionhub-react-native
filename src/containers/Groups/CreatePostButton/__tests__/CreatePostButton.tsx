import React from 'react';
import { useQuery } from '@apollo/react-hooks';
import { fireEvent, flushMicrotasksQueue } from 'react-native-testing-library';

import { renderWithContext } from '../../../../../testUtils';
import { GLOBAL_COMMUNITY_ID } from '../../../../constants';
import {
  FeedItemSubjectTypeEnum,
  PermissionEnum,
  PostTypeEnum,
} from '../../../../../__generated__/globalTypes';
import { GET_MY_COMMUNITY_PERMISSION_QUERY } from '../queries';
import { navigatePush } from '../../../../actions/navigation';
import { CREATE_POST_SCREEN } from '../../CreatePostScreen';
import { CreatePostButton } from '..';

jest.mock('../../../../actions/navigation', () => ({
  navigatePush: jest.fn(() => ({ type: 'navigatePush' })),
}));
jest.mock('../../../../utils/hooks/useAnalytics');
jest.mock('../../../../selectors/people');

const refreshItems = jest.fn();
const mockCommunityId = '1';
const props = {
  refreshItems,
  communityId: mockCommunityId,
};
const initialState = {
  auth: { person: { id: '1' } },
  people: { people: {} },
  drawer: { isOpen: false },
};

it('renders correctly', async () => {
  const { snapshot } = renderWithContext(<CreatePostButton {...props} />, {
    initialState,
  });
  await flushMicrotasksQueue();
  snapshot();
  expect(useQuery).toHaveBeenCalledWith(GET_MY_COMMUNITY_PERMISSION_QUERY, {
    variables: { id: mockCommunityId, myId: '1' },
  });
});

it('renders correctly for admin', async () => {
  const { snapshot } = renderWithContext(<CreatePostButton {...props} />, {
    initialState,
    mocks: {
      CommunityPermission: () => ({ permission: PermissionEnum.admin }),
    },
  });
  await flushMicrotasksQueue();
  snapshot();
  expect(useQuery).toHaveBeenCalledWith(GET_MY_COMMUNITY_PERMISSION_QUERY, {
    variables: { id: mockCommunityId, myId: '1' },
  });
});
it('renders correctly for owner', async () => {
  const { snapshot } = renderWithContext(<CreatePostButton {...props} />, {
    initialState,
    mocks: {
      CommunityPermission: () => ({ permission: PermissionEnum.owner }),
    },
  });
  await flushMicrotasksQueue();
  snapshot();
  expect(useQuery).toHaveBeenCalledWith(GET_MY_COMMUNITY_PERMISSION_QUERY, {
    variables: { id: mockCommunityId, myId: '1' },
  });
});

it('renders correctly with type', () => {
  renderWithContext(
    <CreatePostButton {...props} type={FeedItemSubjectTypeEnum.STORY} />,
    { initialState },
  ).snapshot();
});

it('does not render for Global Community', () => {
  renderWithContext(
    <CreatePostButton {...props} communityId={GLOBAL_COMMUNITY_ID} />,
    { initialState },
  ).snapshot();
});

it('does not render for step', () => {
  renderWithContext(
    <CreatePostButton {...props} type={FeedItemSubjectTypeEnum.STEP} />,
    { initialState },
  ).snapshot();
});

it('does not render announcement', async () => {
  const { snapshot } = renderWithContext(
    <CreatePostButton {...props} type={FeedItemSubjectTypeEnum.ANNOUNCEMENT} />,
    { initialState },
  );
  await flushMicrotasksQueue();
  snapshot();
});

it('does render announcement for owner', async () => {
  const { snapshot } = renderWithContext(
    <CreatePostButton {...props} type={FeedItemSubjectTypeEnum.ANNOUNCEMENT} />,
    {
      initialState,
      mocks: {
        CommunityPermission: () => ({ permission: PermissionEnum.admin }),
      },
    },
  );
  await flushMicrotasksQueue();
  snapshot();
});

it('onPress opens modal', async () => {
  const {
    getByTestId,
    recordSnapshot,
    diffSnapshot,
  } = renderWithContext(<CreatePostButton {...props} />, { initialState });

  await flushMicrotasksQueue();
  recordSnapshot();

  fireEvent.press(getByTestId('CreatePostButton'));
  diffSnapshot();
});

it('onPress navigates for PRAYER_REQUEST ', async () => {
  const { getByTestId } = renderWithContext(
    <CreatePostButton
      {...props}
      type={FeedItemSubjectTypeEnum.PRAYER_REQUEST}
    />,
    { initialState },
  );

  await flushMicrotasksQueue();

  fireEvent.press(getByTestId('CreatePostButton'));
  expect(navigatePush).toHaveBeenLastCalledWith(CREATE_POST_SCREEN, {
    communityId: mockCommunityId,
    postType: PostTypeEnum.prayer_request,
  });
});

it('onPress navigates for QUESTION ', async () => {
  const { getByTestId } = renderWithContext(
    <CreatePostButton {...props} type={FeedItemSubjectTypeEnum.QUESTION} />,
    { initialState },
  );

  await flushMicrotasksQueue();

  fireEvent.press(getByTestId('CreatePostButton'));
  expect(navigatePush).toHaveBeenLastCalledWith(CREATE_POST_SCREEN, {
    communityId: mockCommunityId,
    postType: PostTypeEnum.question,
  });
});

it('onPress navigates for STORY ', async () => {
  const { getByTestId } = renderWithContext(
    <CreatePostButton {...props} type={FeedItemSubjectTypeEnum.STORY} />,
    { initialState },
  );

  await flushMicrotasksQueue();

  fireEvent.press(getByTestId('CreatePostButton'));
  expect(navigatePush).toHaveBeenLastCalledWith(CREATE_POST_SCREEN, {
    communityId: mockCommunityId,
    postType: PostTypeEnum.story,
  });
});

it('onPress navigates for HELP_REQUEST ', async () => {
  const { getByTestId } = renderWithContext(
    <CreatePostButton {...props} type={FeedItemSubjectTypeEnum.HELP_REQUEST} />,
    { initialState },
  );

  await flushMicrotasksQueue();

  fireEvent.press(getByTestId('CreatePostButton'));
  expect(navigatePush).toHaveBeenLastCalledWith(CREATE_POST_SCREEN, {
    communityId: mockCommunityId,
    postType: PostTypeEnum.help_request,
  });
});

it('onPress navigates for THOUGHT ', async () => {
  const { getByTestId } = renderWithContext(
    <CreatePostButton {...props} type={FeedItemSubjectTypeEnum.THOUGHT} />,
    { initialState },
  );

  await flushMicrotasksQueue();

  fireEvent.press(getByTestId('CreatePostButton'));
  expect(navigatePush).toHaveBeenLastCalledWith(CREATE_POST_SCREEN, {
    communityId: mockCommunityId,
    postType: PostTypeEnum.thought,
  });
});

it('onPress navigates for ANNOUNCEMENT ', async () => {
  const { getByTestId } = renderWithContext(
    <CreatePostButton {...props} type={FeedItemSubjectTypeEnum.ANNOUNCEMENT} />,
    {
      initialState,
      mocks: {
        CommunityPermission: () => ({ permission: PermissionEnum.admin }),
      },
    },
  );

  await flushMicrotasksQueue();

  fireEvent.press(getByTestId('CreatePostButton'));
  expect(navigatePush).toHaveBeenLastCalledWith(CREATE_POST_SCREEN, {
    communityId: mockCommunityId,
    postType: PostTypeEnum.announcement,
  });
});
