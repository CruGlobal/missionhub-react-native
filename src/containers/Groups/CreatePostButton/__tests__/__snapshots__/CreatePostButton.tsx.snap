// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not render for Global Community 1`] = `null`;

exports[`onPress opens modal 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -16,10 +16,523 @@
        \\"shadowOpacity\\": 0.3,
        \\"shadowRadius\\": 2,
      }
    }
  >
+   <Modal
+     animationType=\\"slide\\"
+     hardwareAccelerated={false}
+     transparent={true}
+     visible={true}
+   >
+     <View
+       style={
+         Object {
+           \\"alignItems\\": \\"center\\",
+           \\"backgroundColor\\": \\"rgba(0,0,0,0.5)\\",
+           \\"flex\\": 1,
+           \\"justifyContent\\": \\"center\\",
+         }
+       }
+     >
+       <View
+         style={
+           Array [
+             Object {
+               \\"alignItems\\": \\"center\\",
+               \\"backgroundColor\\": \\"#ffffff\\",
+               \\"borderRadius\\": 14,
+               \\"justifyContent\\": \\"space-around\\",
+               \\"padding\\": 14,
+               \\"paddingVertical\\": 50,
+               \\"width\\": \\"80%\\",
+             },
+             Object {
+               \\"flex\\": 0.5,
+             },
+           ]
+         }
+       >
+         <View
+           style={
+             Array [
+               Object {
+                 \\"width\\": \\"100%\\",
+               },
+               Object {
+                 \\"flexDirection\\": \\"row\\",
+                 \\"justifyContent\\": \\"flex-end\\",
+               },
+             ]
+           }
+         >
+           <SvgMock
+             onPress={[Function closeModal]}
+             style={
+               Object {
+                 \\"color\\": \\"#B4B6BA\\",
+                 \\"justifyContent\\": \\"flex-end\\",
+                 \\"position\\": \\"absolute\\",
+                 \\"right\\": -5,
+                 \\"top\\": -40,
+               }
+             }
+             testID=\\"CloseButton\\"
+           />
+         </View>
+         <Text
+           style={
+             Object {
+               \\"color\\": \\"#505256\\",
+               \\"fontFamily\\": \\"SourceSansPro-Regular\\",
+               \\"fontSize\\": 14,
+               \\"lineHeight\\": 20,
+               \\"paddingBottom\\": 20,
+               \\"textAlign\\": \\"center\\",
+             }
+           }
+         >
+           Choose a Post Type
+         </Text>
+         <View
+           accessible={true}
+           focusable={true}
+           isTVSelectable={true}
+           onClick={[Function bound touchableHandlePress]}
+           onResponderGrant={[Function bound touchableHandleResponderGrant]}
+           onResponderMove={[Function bound touchableHandleResponderMove]}
+           onResponderRelease={[Function bound touchableHandleResponderRelease]}
+           onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
+           onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
+           onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
+           style={
+             Object {
+               \\"opacity\\": 1,
+             }
+           }
+           testID=\\"STORYButton\\"
+         >
+           <View
+             style={
+               Array [
+                 Object {
+                   \\"alignItems\\": \\"center\\",
+                   \\"backgroundColor\\": \\"transparent\\",
+                   \\"borderRadius\\": 0,
+                   \\"borderWidth\\": 0,
+                   \\"height\\": 60,
+                   \\"justifyContent\\": \\"center\\",
+                   \\"margin\\": 0,
+                   \\"paddingHorizontal\\": 20,
+                   \\"paddingVertical\\": 10,
+                 },
+                 null,
+                 Array [
+                   Object {
+                     \\"alignItems\\": \\"center\\",
+                     \\"flexDirection\\": \\"row\\",
+                     \\"height\\": 36,
+                     \\"justifyContent\\": \\"space-between\\",
+                     \\"paddingVertical\\": 10,
+                   },
+                   Object {
+                     \\"backgroundColor\\": \\"#7076B5\\",
+                   },
+                   Object {
+                     \\"height\\": 48,
+                   },
+                   null,
+                 ],
+                 Object {
+                   \\"borderRadius\\": 50,
+                 },
+               ]
+             }
+           >
+             <SvgMock
+               color=\\"#ffffff\\"
+               height={24}
+               style={
+                 Object {
+                   \\"marginLeft\\": -10,
+                   \\"marginRight\\": 10,
+                 }
+               }
+               width={24}
+             />
+             <Text
+               style={
+                 Array [
+                   Object {
+                     \\"color\\": \\"#505256\\",
+                   },
+                   Object {
+                     \\"fontFamily\\": \\"SourceSansPro-Regular\\",
+                   },
+                   Object {
+                     \\"color\\": \\"#ffffff\\",
+                     \\"fontSize\\": 14,
+                     \\"lineHeight\\": 20,
+                   },
+                 ]
+               }
+             >
+               God Story
+             </Text>
+           </View>
+         </View>
+         <View
+           accessible={true}
+           focusable={true}
+           isTVSelectable={true}
+           onClick={[Function bound touchableHandlePress]}
+           onResponderGrant={[Function bound touchableHandleResponderGrant]}
+           onResponderMove={[Function bound touchableHandleResponderMove]}
+           onResponderRelease={[Function bound touchableHandleResponderRelease]}
+           onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
+           onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
+           onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
+           style={
+             Object {
+               \\"opacity\\": 1,
+             }
+           }
+           testID=\\"PRAYER_REQUESTButton\\"
+         >
+           <View
+             style={
+               Array [
+                 Object {
+                   \\"alignItems\\": \\"center\\",
+                   \\"backgroundColor\\": \\"transparent\\",
+                   \\"borderRadius\\": 0,
+                   \\"borderWidth\\": 0,
+                   \\"height\\": 60,
+                   \\"justifyContent\\": \\"center\\",
+                   \\"margin\\": 0,
+                   \\"paddingHorizontal\\": 20,
+                   \\"paddingVertical\\": 10,
+                 },
+                 null,
+                 Array [
+                   Object {
+                     \\"alignItems\\": \\"center\\",
+                     \\"flexDirection\\": \\"row\\",
+                     \\"height\\": 36,
+                     \\"justifyContent\\": \\"space-between\\",
+                     \\"paddingVertical\\": 10,
+                   },
+                   Object {
+                     \\"backgroundColor\\": \\"#A97398\\",
+                   },
+                   Object {
+                     \\"height\\": 48,
+                   },
+                   null,
+                 ],
+                 Object {
+                   \\"borderRadius\\": 50,
+                 },
+               ]
+             }
+           >
+             <SvgMock
+               color=\\"#ffffff\\"
+               height={24}
+               style={
+                 Object {
+                   \\"marginLeft\\": -10,
+                   \\"marginRight\\": 10,
+                 }
+               }
+               width={24}
+             />
+             <Text
+               style={
+                 Array [
+                   Object {
+                     \\"color\\": \\"#505256\\",
+                   },
+                   Object {
+                     \\"fontFamily\\": \\"SourceSansPro-Regular\\",
+                   },
+                   Object {
+                     \\"color\\": \\"#ffffff\\",
+                     \\"fontSize\\": 14,
+                     \\"lineHeight\\": 20,
+                   },
+                 ]
+               }
+             >
+               Prayer Request
+             </Text>
+           </View>
+         </View>
+         <View
+           accessible={true}
+           focusable={true}
+           isTVSelectable={true}
+           onClick={[Function bound touchableHandlePress]}
+           onResponderGrant={[Function bound touchableHandleResponderGrant]}
+           onResponderMove={[Function bound touchableHandleResponderMove]}
+           onResponderRelease={[Function bound touchableHandleResponderRelease]}
+           onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
+           onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
+           onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
+           style={
+             Object {
+               \\"opacity\\": 1,
+             }
+           }
+           testID=\\"QUESTIONButton\\"
+         >
+           <View
+             style={
+               Array [
+                 Object {
+                   \\"alignItems\\": \\"center\\",
+                   \\"backgroundColor\\": \\"transparent\\",
+                   \\"borderRadius\\": 0,
+                   \\"borderWidth\\": 0,
+                   \\"height\\": 60,
+                   \\"justifyContent\\": \\"center\\",
+                   \\"margin\\": 0,
+                   \\"paddingHorizontal\\": 20,
+                   \\"paddingVertical\\": 10,
+                 },
+                 null,
+                 Array [
+                   Object {
+                     \\"alignItems\\": \\"center\\",
+                     \\"flexDirection\\": \\"row\\",
+                     \\"height\\": 36,
+                     \\"justifyContent\\": \\"space-between\\",
+                     \\"paddingVertical\\": 10,
+                   },
+                   Object {
+                     \\"backgroundColor\\": \\"#FFA178\\",
+                   },
+                   Object {
+                     \\"height\\": 48,
+                   },
+                   null,
+                 ],
+                 Object {
+                   \\"borderRadius\\": 50,
+                 },
+               ]
+             }
+           >
+             <SvgMock
+               color=\\"#ffffff\\"
+               height={24}
+               style={
+                 Object {
+                   \\"marginLeft\\": -10,
+                   \\"marginRight\\": 10,
+                 }
+               }
+               width={24}
+             />
+             <Text
+               style={
+                 Array [
+                   Object {
+                     \\"color\\": \\"#505256\\",
+                   },
+                   Object {
+                     \\"fontFamily\\": \\"SourceSansPro-Regular\\",
+                   },
+                   Object {
+                     \\"color\\": \\"#ffffff\\",
+                     \\"fontSize\\": 14,
+                     \\"lineHeight\\": 20,
+                   },
+                 ]
+               }
+             >
+               Spiritual Question
+             </Text>
+           </View>
+         </View>
+         <View
+           accessible={true}
+           focusable={true}
+           isTVSelectable={true}
+           onClick={[Function bound touchableHandlePress]}
+           onResponderGrant={[Function bound touchableHandleResponderGrant]}
+           onResponderMove={[Function bound touchableHandleResponderMove]}
+           onResponderRelease={[Function bound touchableHandleResponderRelease]}
+           onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
+           onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
+           onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
+           style={
+             Object {
+               \\"opacity\\": 1,
+             }
+           }
+           testID=\\"HELP_REQUESTButton\\"
+         >
+           <View
+             style={
+               Array [
+                 Object {
+                   \\"alignItems\\": \\"center\\",
+                   \\"backgroundColor\\": \\"transparent\\",
+                   \\"borderRadius\\": 0,
+                   \\"borderWidth\\": 0,
+                   \\"height\\": 60,
+                   \\"justifyContent\\": \\"center\\",
+                   \\"margin\\": 0,
+                   \\"paddingHorizontal\\": 20,
+                   \\"paddingVertical\\": 10,
+                 },
+                 null,
+                 Array [
+                   Object {
+                     \\"alignItems\\": \\"center\\",
+                     \\"flexDirection\\": \\"row\\",
+                     \\"height\\": 36,
+                     \\"justifyContent\\": \\"space-between\\",
+                     \\"paddingVertical\\": 10,
+                   },
+                   Object {
+                     \\"backgroundColor\\": \\"#FD726D\\",
+                   },
+                   Object {
+                     \\"height\\": 48,
+                   },
+                   null,
+                 ],
+                 Object {
+                   \\"borderRadius\\": 50,
+                 },
+               ]
+             }
+           >
+             <SvgMock
+               color=\\"#ffffff\\"
+               height={24}
+               style={
+                 Object {
+                   \\"marginLeft\\": -10,
+                   \\"marginRight\\": 10,
+                 }
+               }
+               width={24}
+             />
+             <Text
+               style={
+                 Array [
+                   Object {
+                     \\"color\\": \\"#505256\\",
+                   },
+                   Object {
+                     \\"fontFamily\\": \\"SourceSansPro-Regular\\",
+                   },
+                   Object {
+                     \\"color\\": \\"#ffffff\\",
+                     \\"fontSize\\": 14,
+                     \\"lineHeight\\": 20,
+                   },
+                 ]
+               }
+             >
+               Care Request
+             </Text>
+           </View>
+         </View>
+         <View
+           accessible={true}
+           focusable={true}
+           isTVSelectable={true}
+           onClick={[Function bound touchableHandlePress]}
+           onResponderGrant={[Function bound touchableHandleResponderGrant]}
+           onResponderMove={[Function bound touchableHandleResponderMove]}
+           onResponderRelease={[Function bound touchableHandleResponderRelease]}
+           onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
+           onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
+           onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
+           style={
+             Object {
+               \\"opacity\\": 1,
+             }
+           }
+           testID=\\"THOUGHTButton\\"
+         >
+           <View
+             style={
+               Array [
+                 Object {
+                   \\"alignItems\\": \\"center\\",
+                   \\"backgroundColor\\": \\"transparent\\",
+                   \\"borderRadius\\": 0,
+                   \\"borderWidth\\": 0,
+                   \\"height\\": 60,
+                   \\"justifyContent\\": \\"center\\",
+                   \\"margin\\": 0,
+                   \\"paddingHorizontal\\": 20,
+                   \\"paddingVertical\\": 10,
+                 },
+                 null,
+                 Array [
+                   Object {
+                     \\"alignItems\\": \\"center\\",
+                     \\"flexDirection\\": \\"row\\",
+                     \\"height\\": 36,
+                     \\"justifyContent\\": \\"space-between\\",
+                     \\"paddingVertical\\": 10,
+                   },
+                   Object {
+                     \\"backgroundColor\\": \\"#B4B6BA\\",
+                   },
+                   Object {
+                     \\"height\\": 48,
+                   },
+                   null,
+                 ],
+                 Object {
+                   \\"borderRadius\\": 50,
+                 },
+               ]
+             }
+           >
+             <SvgMock
+               color=\\"#ffffff\\"
+               height={24}
+               style={
+                 Object {
+                   \\"marginLeft\\": -10,
+                   \\"marginRight\\": 10,
+                 }
+               }
+               width={24}
+             />
+             <Text
+               style={
+                 Array [
+                   Object {
+                     \\"color\\": \\"#505256\\",
+                   },
+                   Object {
+                     \\"fontFamily\\": \\"SourceSansPro-Regular\\",
+                   },
+                   Object {
+                     \\"color\\": \\"#ffffff\\",
+                     \\"fontSize\\": 14,
+                     \\"lineHeight\\": 20,
+                   },
+                 ]
+               }
+             >
+               What's on Your Mind
+             </Text>
+           </View>
+         </View>
+       </View>
+     </View>
+   </Modal>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function bound touchableHandlePress]}"
`;

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderBottomColor": "#ECEEF2",
      "borderBottomWidth": 1,
      "elevation": 4,
      "marginBottom": 5,
      "marginTop": -5,
      "paddingTop": 5,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.3,
      "shadowRadius": 2,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="CreatePostButton"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderRadius": 0,
            "borderWidth": 0,
            "height": 60,
            "justifyContent": "center",
            "margin": 0,
            "paddingHorizontal": 20,
            "paddingVertical": 10,
          },
          null,
          Object {
            "alignItems": "center",
            "backgroundColor": "#ECEEF2",
            "borderRadius": 20,
            "flexDirection": "row",
            "height": 36,
            "justifyContent": "flex-start",
            "marginHorizontal": 20,
            "marginVertical": 20,
          },
          null,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
            Array [
              Object {
                "borderRadius": 18,
                "height": 36,
                "width": 36,
              },
              Object {
                "backgroundColor": "rgb(199, 163, 67)",
              },
              Object {
                "marginLeft": -15,
              },
            ],
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#505256",
              },
              Object {
                "fontFamily": "SourceSansPro-Regular",
              },
              Array [
                Object {
                  "color": "#ffffff",
                },
                Object {
                  "fontSize": 20,
                  "fontWeight": "300",
                },
              ],
            ]
          }
        >
          -
        </Text>
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#505256",
            },
            Object {
              "fontFamily": "SourceSansPro-Regular",
            },
            Object {
              "fontSize": 14,
              "height": 18,
              "paddingLeft": 10,
            },
          ]
        }
      >
        Post to community...
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders correctly with type 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderBottomColor": "#ECEEF2",
      "borderBottomWidth": 1,
      "elevation": 4,
      "marginBottom": 5,
      "marginTop": -5,
      "paddingTop": 5,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.3,
      "shadowRadius": 2,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="CreatePostButton"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderRadius": 0,
            "borderWidth": 0,
            "height": 60,
            "justifyContent": "center",
            "margin": 0,
            "paddingHorizontal": 20,
            "paddingVertical": 10,
          },
          null,
          Object {
            "alignItems": "center",
            "backgroundColor": "#ECEEF2",
            "borderRadius": 20,
            "flexDirection": "row",
            "height": 36,
            "justifyContent": "flex-start",
            "marginHorizontal": 20,
            "marginVertical": 20,
          },
          null,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
            Array [
              Object {
                "borderRadius": 18,
                "height": 36,
                "width": 36,
              },
              Object {
                "backgroundColor": "rgb(199, 163, 67)",
              },
              Object {
                "marginLeft": -15,
              },
            ],
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#505256",
              },
              Object {
                "fontFamily": "SourceSansPro-Regular",
              },
              Array [
                Object {
                  "color": "#ffffff",
                },
                Object {
                  "fontSize": 20,
                  "fontWeight": "300",
                },
              ],
            ]
          }
        >
          -
        </Text>
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#505256",
            },
            Object {
              "fontFamily": "SourceSansPro-Regular",
            },
            Object {
              "fontSize": 14,
              "height": 18,
              "paddingLeft": 10,
            },
          ]
        }
      >
        Share a God Story
      </Text>
    </View>
  </View>
</View>
`;
